src/storage/src/source/mysql/statistics.rs

```diff
@@ -0,0 +1,152 @@
+            let mut offset_committed = None;
+
+            let upstream_stream = async_stream::stream!({
+                // TODO(guswynn): make configurable (at the same time as the pg one).
+                let mut interval = tokio::time::interval(std::time::Duration::from_secs(10));
+                loop {
+                    interval.tick().await;
+
+                    let gtid_executed =
+                        query_sys_var(&mut stats_conn, "global.gtid_executed").await?;
+                    match gtid_set_frontier(&gtid_executed) {
+                        Ok(frontier) => yield Ok(frontier),
+                        Err(err) => {
+                            // We don't translate this into a definite error like in snapshotting, but we will
+                            // restart the source.
+                            yield Err::<_, TransientError>(err.into())
+                        }
+                    }
+                }
+            });
+            tokio::pin!(upstream_stream);
+
+            loop {
+                tokio::select! {
+                    Some(upstream_frontier) = upstream_stream.next() => {
+                        offset_known = Some(aggregate_mysql_frontier(upstream_frontier?));
+
+                    },
+                    Some(committed_frontier) = resume_uppers.next() => {
+                        offset_committed = Some(aggregate_mysql_frontier(committed_frontier));
+                    },
+                    else => break
+                };
+
+                if let (Some(offset_known), Some(offset_committed)) =
+                    (offset_known, offset_committed)
+                {
+                    stats_output
+                        .give(
+                            &stats_cap[0],
+                            ProgressStatisticsUpdate {
+                                offset_known,
+                                offset_committed,
+                            },
+                        )
+                        .await;
+                }
+            }
+
+            Ok(())
+        })
+    });
+
+    (
+        stats_stream,
+        transient_errors.map(ReplicationError::from),
+        button.press_on_drop(),
+    )
+}
+
+/// Aggregate a mysql frontier into single number representing the
+/// _number of transactions_ it represents.
+fn aggregate_mysql_frontier(frontier: Antichain<GtidPartition>) -> u64 {
+    let mut progress_stat = 0;
+    for ts in frontier.iter() {
+        if let Some(_uuid) = ts.interval().singleton() {
+            // We assume source id's don't disappear once they appear.
+            let ts = match ts.timestamp() {
+                GtidState::Absent => 0,
+                GtidState::Active(id) => id.get(),
+            };
+            progress_stat += ts;
+        }
+    }
+    // Txid's in mysql start at 1, so we subtract 1 from the _frontier_
+    // to get the _number of transactions_.
+    progress_stat.saturating_sub(1)
```

Comment:
would this subtraction need to be once-per-non-zero-source-id?

It might be better to do this above:
```
for ts in frontier.iter() {
        if let Some(_uuid) = ts.interval().singleton() {
            // We assume source id's don't disappear once they appear.
            let ts = match ts.timestamp() {
                GtidState::Absent => 0,
                GtidState::Active(id) => id.get().saturating_sub(1),
            };
            progress_stat += ts;
        }
    }
```